addons/mail/static/src/core/common/persona_model.js

```diff
@@ -164,6 +164,15 @@ export class Persona extends Record {
         }
         this.im_status = newStatus;
     }
+
+    /**
+     * Return the name to display when shown in the given channel.
+     *
+     * @param {import("models").Thread} channel
+     */
+    getNameForChannel(channel) {
```

Comment:
I would see this method on thread rather than on persona as most of the condition/override is channel specific.

Note: we had this before the last big refactoring (https://github.com/odoo/odoo/blob/16.0/addons/mail/static/src/models/thread.js#L651), 1.5 years later still having to "fix" "simplifications"... This is a bit frustrating.

Reply:
Reactions are not linked to members so I have no choice. Also, even if we retrieved the member via the `message.thread` relation (which is a long shot) there is still the possibility that the author left the chat, we still would want to use `livechat_username` in this case. So I think we should keep this on `persona_model`.

Reply:
We don't need member to move it to thread, this method just has 2 params thread and persona, one will be given as param and the other as `this`, I'm just suggesting we invert them.

But other places (such as member) can call this new method to get the proper name.